### ISSUE TITLE
Support FastLED and NeoPixelBus on the RP2040

### ISSUE THREAD

**Issue by ArthurGuy** [esphome/feature-requests#1998]
**Describe the problem you have/What new integration you would like**
I'd like it if ESPHome could support FastLED and/or NeoPixelBus on the RP2040 board

**Please describe your use case for this integration and alternatives you've tried:**
There are several [RP2040 based boards](https://shop.pimoroni.com/products/plasma-stick-2040-w) designed for controlling NeoPixels or other WS2812 led strips, it would be great if they could be run with ESPHome

**Additional context**

**Comment by Sese-Schneider** [comment]
+1 on this.
I do also see a lot of use cases for this now and in the future, especially as the RP Pico W is a very cheap board.

As of https://github.com/FastLED/FastLED/commit/aee28fc358ce0a4c26aa55e12b1ac0e1843d7314 FastLED support for the RP2040 board is added to FastLED, therefore I assume the addition of the library could be a straight forward task.

cc: @jesserockz as maintainer of the RP2040 integration

**Comment by hnrkp** [comment]
Seems the latest version of FastLED (3.5.0) didn't ship the full rp2040 support (platform missing, headers seems to not include the pico). However, I did build it against master (changed the codegen thing to cg.add_library("FastLED", None, "https://github.com/fastled/FastLED.git")) and it boots and seems to work.

**Comment by gretel** [comment]
[#4257](https://github.com/esphome/esphome/pull/4257) werks for me:

```yaml
external_components:
  - source: github://pr#4257
    components: [fastled]
    refresh: always
```

```yaml
light:
  - platform: fastled
...
```

**Comment by ArthurGuy** [comment]
@gretel This is now compiling but I haven't been able to get it to work, have you?

**Comment by gretel** [comment]
> @gretel This is now compiling but I haven't been able to get it to work, have you?

yes it's werking. tho, looks like with more than one led only one gets it's settings changed on call. sort of `fifo` 😄

**Comment by ArthurGuy** [comment]
It's all working now, some other config issue causing problems. Thanks!

**Comment by nielsnl68** [comment]
Thanks for testing my changes. @gretel and @ArthurGuy. Could you please report this in the PR as well with the config you are using.

**Comment by nielsnl68** [comment]
> > @gretel This is now compiling but I haven't been able to get it to work, have you?
> 
> yes it's werking. tho, looks like with more than one led only one gets it's settings changed on call. sort of `fifo` 😄

could you explain more about this and maybe could you make a vid about it that i can watch?
